web/source/osk/banner.ts

```diff
@@ -3,47 +3,76 @@ namespace com.keyman.osk {
   // Base class for a banner above the keyboard in the OSK
 
   export abstract class Banner {
-    private _Visible: boolean;
-    private _Enabled: boolean;
-    private _Height: number; // pixels
-    protected div: HTMLDivElement;
+    private _enabled: boolean;
+    private _height: number; // pixels
+    private div: HTMLDivElement;
 
     public DEFAULT_HEIGHT: number = 40; // pixels
 
-    public get height():number {
-      return this._Height;
+    /**
+     * Function     height
+     * Scope        Public
+     * @returns     {number} height in pixels
+     * Description  Returns the height of the banner in pixels
+     */
+    public get height(): number {
+      return this._height;
     }
 
-    public set height(height:number) {
-      this._Height = height;
-      if (this.div) {
-        let ds = this.div.style;
-        ds.height = (height > 0) ? height + 'px' : '0px';
-      }
+    /**
+     * Function     height
+     * Scope        Public
+     * @param       {number} height   the height in pixels
+     * Description  Sets the height of the banner in pixels. If a negative height is given, use a default height.
+     *              Also updates the banner styling.
+     */
+    public set height(height: number) {
+      this._height = (height > 0) ?  height : this.DEFAULT_HEIGHT;
+      this.update();
     }
 
-    public get visible():boolean {
-      return this._Visible;
+    /**
+     * Function     enable
+     * Scope        Public
+     * @return      {boolean} true if the banner is enabled
+     * Description  Returns whether the banner is enabled or not
+     */
+    public get enable(): boolean {
+      return this._enabled;
     }
 
-    public set visible(visible: boolean) {
-      this._Visible = visible;
-
-      if (this.div) {
-        let ds = this.div.style;
-        ds.display=(this._Visible) ? 'block': 'none';
-      }
+    /**
+     * Function     visible
```

Comment:
Incorrect function name :)

Reply:
What the different between `enabled` and `visible` ... I know `visible` is gone, so I guess I'm asking what's the difference between `enabled` and `height > 0`?)

Reply:
atm, `enabled` is a TODO for BannerManager to control.

Reply:
So since it's not used, I've removed the `enabled` property too.